tests/checkpointing/test_trainer_checkpoint.py

```diff
@@ -84,3 +85,25 @@ def validation_step(self, batch, batch_idx):
             assert best_model_path.endswith(f"epoch=0{idx}.ckpt")
         else:
             assert f"epoch={idx + 1}" in best_model_path
+
+
+def test_accumulated_gradient_batches_with_resume_from_checkpoint(tmpdir):
+    """
+    This test validates that accumulated gradient properly is recomputed and reset on the trainer.
+    """
+
+    train_data = DataLoader(RandomDataset(32, 64), num_workers=0)
+    val_data = DataLoader(RandomDataset(32, 64), num_workers=0)
+
+    cb = ModelCheckpoint(dirpath=tmpdir, save_last=True)
+    model = BoringModel()
+    trainer_kwargs = dict(
+        max_epochs=1, accumulate_grad_batches={0: 2}, callbacks=cb, limit_train_batches=1, limit_val_batches=0
+    )
+    trainer = Trainer(**trainer_kwargs)
+    trainer.fit(model, train_data, val_data)
+
+    trainer_kwargs['max_epochs'] = 2
+    trainer_kwargs['resume_from_checkpoint'] = cb.last_model_path
+    trainer = Trainer(**trainer_kwargs)
+    trainer.fit(model, train_data, val_data)
```

Comment:
```suggestion
def test_accumulated_gradient_batches_with_resume_from_checkpoint(tmpdir):
    """
    This test validates that accumulated gradient properly is recomputed and reset on the trainer.
    """
    cb = ModelCheckpoint(dirpath=tmpdir, save_last=True)
    model = BoringModel()
    trainer_kwargs = dict(
        max_epochs=1, accumulate_grad_batches={0: 2}, callbacks=cb, limit_train_batches=1, limit_val_batches=0
    )
    trainer = Trainer(**trainer_kwargs)
    trainer.fit(model)

    trainer_kwargs['max_epochs'] = 2
    trainer_kwargs['resume_from_checkpoint'] = cb.last_model_path
    trainer = Trainer(**trainer_kwargs)
    trainer.fit(model)
```

Reply:
The boringmodel already has dataloaders defined

Reply:
Removed !